examples/dp-sgd-mnist/client.py

```diff
@@ -0,0 +1,154 @@
+                l2_norm_clip=args.l2_norm_clip,
+                noise_multiplier=args.noise_multiplier,
+                num_microbatches=args.microbatches,
+                learning_rate=args.learning_rate)
+            # Compute vector of per-example loss rather than its mean over a minibatch.
+            loss = tf.keras.losses.CategoricalCrossentropy(
+                from_logits=True, reduction=tf.losses.Reduction.NONE)
+        else:
+            optimizer = tf.keras.optimizers.SGD(learning_rate=args.learning_rate)
+            loss = tf.keras.losses.CategoricalCrossentropy(from_logits=True)
+
+        # Compile model with Keras
+        model.compile(optimizer=optimizer, loss=loss, metrics=['accuracy'])
+
+    def get_parameters(self):
+        """Get parameters of the local model."""
+        raise Exception("Not implemented (server-side parameter initialization)")
+
+    def fit(self, parameters, config):
+        """Train parameters on the locally held training set."""
+        # Update local model parameters
+        global PRIVACY_LOSS
+        if self.dpsgd:
+            privacy_spent = common.compute_epsilon(
+                self.local_epochs,
+                len(self.x_train),
+                self.batch_size,
+                self.noise_multiplier,
+            )
+            PRIVACY_LOSS += privacy_spent
+
+        self.model.set_weights(parameters)
+        # Train the model
+        self.model.fit(
+            self.x_train, self.y_train,
+            epochs=self.local_epochs,
+            batch_size=self.batch_size)
+
+        return self.model.get_weights(), len(self.x_train), {}
+
+    def evaluate(self, parameters, config):
+        """Evaluate parameters on the locally held test set."""
+
+        # Update local model with global parameters
+        self.model.set_weights(parameters)
+
+        # Evaluate global model parameters on the local test data and return results
+        loss, accuracy = self.model.evaluate(self.x_test, self.y_test)
+        num_examples_test = len(self.x_test)
+        return loss, num_examples_test, {"accuracy": accuracy}
+
+
+def main(args) -> None:
+    # Load and compile Keras model
+    model = common.create_cnn_model()
+
+    # Load a subset of CIFAR-10 to simulate the local data partition
+    (x_train, y_train), (x_test, y_test) = common.load(
+        args.num_clients)[args.partition]
+
+    # drop samples to form exact batches for dpsgd
+    # this is necessary since dpsgd is sensitive to uneven batches
+    # due to microbatches
+    if args.dpsgd and x_train.shape[0] % args.batch_size != 0:
+        drop_num = x_train.shape[0] % args.batch_size
+        x_train = x_train[:-drop_num]
+        y_train = y_train[:-drop_num]
+
+    # Start Flower client
+    client = MnistClient(model, x_train, y_train, x_test, y_test, args)
+    fl.client.start_numpy_client("[::]:8080", client=client)
+    if args.dpsgd:
+        print("Privacy Lost: ", PRIVACY_LOSS)
```

Comment:
Should the log message change here?
```suggestion
        print("Privacy Loss: ", PRIVACY_LOSS)
```